R/eval_forecasts.R

```diff
@@ -284,26 +356,71 @@ eval_forecasts <- function(data,
   dat[, names(dat)[grepl("sampl_", names(dat))] := NULL]
   dat[, c("sharpness", "bias", "dss", "crps") := NULL]
 
-
   # merge with previous data
   merge_cols = colnames(dat)[!colnames(dat) %in% c("pit_p_val", "pit_sd")]
   res <- merge(data, dat, by = merge_cols)
 
-  if (prediction_type == "continuous") {
-    scores <- c("sharpness", "bias", "dss", "crps", "log_score", "pit_p_val")
-  } else {
-    scores <- c("sharpness", "bias", "dss", "crps", "pit_p_val", "pit_sd")
-  }
-
-  # aggregate scores so there is only one score per observed value
-  res <- res[, lapply(.SD, mean, na.rm = TRUE),
-             .SDcols = scores,
+  # @ SAM
```

Comment:
I think if I understand this properly we want to filter out duplicated scores? Can you just reiterate why scores are duplicated?

Reply:
So say I get one score per country or whatever is defined in the by argument. Then I can either still return a data.frame that has the exact same dimensions as the input data.frame. Or I can alternatively just take the unique value over the country and just return one value per group in `by`. So the information should be the same, it is just whether the data.frame returned should have the same dimensions as the input data.frame with scores potentially repeated. I slightly lean towards that solution.

Reply:
so e.g. if you have 1000 samples to predict one true observation, then there will either be the same score for every one of these samples. Or you could have the sample column removed and just give one score for the one true observation.

Reply:
ah okay. I would assume you return scores in the format of by. Why would they ever want duplicates?

Reply:
Hm maybe there is a use case for still having all your individual samples and just add scores? unsure

Reply:
it doesn't seem like a common one and if that was what you wanted can't you just join